src/components/AddPresetAsset.tsx

```diff
@@ -0,0 +1,180 @@
+    "allAssets",
+    "settings",
+    "untrustedAssets",
+  );
+  const [presetAssets, setPresetAssets] = useState<presetAsset[]>([]);
+  const [checkedAssets, setCheckedAssets] = useState<{
+    [key: string]: boolean;
+  }>({});
+  const [isValidating, setIsValidating] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const history = useHistory();
+
+  useEffect(() => {
+    setPresetAssets(getPresetAssets(allAssets.data));
+  }, [allAssets]);
+
+  useEffect(() => {
+    if (untrustedAssets.status === ActionStatus.SUCCESS) {
+      onClose();
+    }
+
+    if (untrustedAssets.errorString) {
+      setErrorMessage(untrustedAssets.errorString);
+    }
+  }, [onClose, untrustedAssets.errorString, untrustedAssets.status]);
+
+  const getAssetId = (asset: presetAsset) =>
+    `${asset.assetCode}:${asset.homeDomain || asset.issuerPublicKey}`;
+
+  const handleOnChangeCheckbox = (asset: presetAsset) => {
+    const updatedCheckedAssets = JSON.parse(JSON.stringify(checkedAssets));
+    const assetId = getAssetId(asset);
+    updatedCheckedAssets[assetId] = !checkedAssets[assetId];
+    setCheckedAssets(updatedCheckedAssets);
+  };
+
+  const handleAddUntrustedAssets = async (assetList: presetAsset[]) => {
+    setErrorMessage("");
+    setIsValidating(true);
+
+    try {
+      const validatedAssetValues = await Promise.all(
+        assetList.map(async (pAsset) => {
+          const { assetCode, issuerPublicKey, homeDomain } = pAsset;
+
+          if (!(homeDomain || issuerPublicKey)) {
+            const errorMsg = `Home domain OR issuer public key is required with asset code ${assetCode}`;
+            log.error({ title: errorMsg });
+            throw new Error(errorMsg);
+          }
+
+          const asset = await getValidatedUntrustedAsset({
+            assetCode,
+            homeDomain,
+            issuerPublicKey,
+            accountBalances: account.data?.balances,
+            networkUrl: getNetworkConfig(settings.pubnet).url,
+          });
+
+          return `${asset.assetCode}:${asset.assetIssuer}`;
+        }),
+      );
+
+      const newSearchQ = searchParam.update(
+        SearchParams.UNTRUSTED_ASSETS,
+        validatedAssetValues,
+      );
+      history.push(newSearchQ);
+    } catch (e) {
+      const errorMsg = getErrorMessage(e);
+      log.error({ title: errorMsg });
+      setErrorMessage(errorMsg);
+      setIsValidating(false);
+      return;
+    }
+
+    setIsValidating(false);
+  };
+
+  const isPending =
+    isValidating || untrustedAssets.status === ActionStatus.PENDING;
+
+  const renderAssetRow = (asset: presetAsset) => {
+    const { homeDomain, issuerPublicKey: assetIssuer } = asset;
+    const assetId = getAssetId(asset);
+    const issuerLink =
+      (homeDomain && `//${homeDomain}/.well-known/stellar.toml`) ||
+      (assetIssuer &&
+        `${getNetworkConfig(settings.pubnet).url.replace(
+          "https:",
+          "",
+        )}/accounts/${assetIssuer}`);
```

Comment:
Could we add a comment here for clarity? 😊 That we are using accounts endpoint data if there is no home domain, correct?

Also, same as earlier about pubnet.

Reply:
Done in https://github.com/stellar/stellar-demo-wallet/pull/206/commits/3a63377862a8265c9bc0e413fcd018ed7a13d9f0.